test/model/message_list_test.dart

```diff
@@ -233,6 +252,27 @@ void main() {
       ..messages.length.equals(200);
   });
 
+
+  test('fetchOlder, recent senders track all the messages', () async {
+    const narrow = CombinedFeedNarrow();
+    await prepare(narrow: narrow);
+    final initialMessages = List.generate(10, (i) => eg.streamMessage(id: 100 + i));
+    await prepareMessages(foundOldest: false, messages: initialMessages);
+
+    final oldMessages = List.generate(10,
+      (i) => eg.streamMessage(id: 90 + i))
+        // Not subscribed to the stream with id 10.
+        ..add(eg.streamMessage(stream: eg.stream(streamId: 10)));
```

Comment:
This last message will have a larger message ID than any of the list's existing messages (the ones in `initialMessages`.

That apparently doesn't break this test, but it is unrealistic and could be confusing. So just specify an ID to avoid that. (And adjust the other IDs to make room; as is, it belongs between 99 and 100.)